From bc29815b1b94282bd17e42f85fa3db39f4dedf91 Mon Sep 17 00:00:00 2001
From: Zaikuo Wang <zaikuo.wang@intel.com>
Date: Tue, 5 May 2015 11:50:44 +0800
Subject: [PATCH 379/462] ov5693: move restore res setting to s_power

originally, res setting would restore and be incorrect
at ov5693_init() function when we try powerup sensor,
because driver has set run_mode and choose correct setting array
before this; otherwise ipanic or other unexpected issue hapen

Change-Id: Iff7d14d3c4c6037eed88cba33801afa96bf0da56
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-34972
Signed-off-by: Zaikuo Wang <zaikuo.wang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/54270
Reviewed-by: Tian Shu Qiu <tian.shu.qiu@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Build: IMIN (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/i2c/ov5693-ecs/ov5693.c             | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
index 3ca28dc..2135828 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
@@ -1317,10 +1317,6 @@ static int ov5693_init(struct v4l2_subdev *sd)
 				"vcm change mode failed\n");
 	}
 
-	/* restore settings */
-	ov5693_res = ov5693_res_preview;
-	N_RES = N_RES_PREVIEW;
-
 	/*change initial focus value for ad5823*/
 	if (dev->vcm == VCM_AD5823) {
 		dev->focus = AD5823_INIT_FOCUS_POS;
@@ -1492,8 +1488,12 @@ static int ov5693_s_power(struct v4l2_subdev *sd, int on)
 		return power_down(sd);
 	else {
 		ret = power_up(sd);
-		if (!ret)
-			return ov5693_init(sd);
+		if (!ret) {
+			ret = ov5693_init(sd);
+			/* restore settings */
+			ov5693_res = ov5693_res_preview;
+			N_RES = N_RES_PREVIEW;
+		}
 	}
 	return ret;
 }
-- 
1.9.1

